lib/daru/dataframe.rb

```diff
@@ -1839,23 +1947,16 @@ def prepare_vector_for_insert name, vector
       end
     end
 
-    def insert_or_modify_row name, vector
-      raise NotImplementedError, "Still can't insert rows in multi index dataframes" \
-        if index.is_a?(MultiIndex)
+    def insert_or_modify_row indexes, vector
+      vector = preprocess_vector vector
 
-      name = name[0]
-      vec = Vector.coerce(vector, name: coerce_name(name), index: @vectors)
+      raise SizeError, 'Vector length should match row length' if
+        vector.size != @vectors.size
 
```

Comment:
Ok. This file wasn't included earlier and `SizeError` wasn't recognized. I'm including it in `dataframe.rb`